src/mgr/ActivePyModules.cc

```diff
@@ -167,20 +169,49 @@ PyObject *ActivePyModules::get_daemon_status_python(
   return f.get();
 }
 
+void ActivePyModules::update_cache_metrics() {
+    auto hit_miss_ratio = ttl_cache.get_hit_miss_ratio();
+    perfcounter->set(l_mgr_cache_hit, hit_miss_ratio.first);
+    perfcounter->set(l_mgr_cache_miss, hit_miss_ratio.second);
+}
+
 PyObject *ActivePyModules::get_python(const std::string &what)
 {
-  PyFormatter f;
+  uint64_t ttl_seconds = g_conf().get_val<uint64_t>("mgr_ttl_cache_expire_seconds");
+  if(ttl_seconds > 0) {
+    ttl_cache.set_ttl(ttl_seconds);
+    try{
+      PyObject* cached = ttl_cache.get(what);
+      update_cache_metrics();
+      return cached;
+    } catch (std::out_of_range& e) {}
+  }
+
+  PyObject *obj = _get_python(what);
+  if(ttl_seconds && ttl_cache.is_allowed(what)) {
+    ttl_cache.insert(what, obj);
+    Py_INCREF(obj);
+  }
+  update_cache_metrics();
+  return obj;
+}
 
+PyObject *ActivePyModules::_get_python(const std::string &what)
+{
+  uint64_t ttl_seconds = g_conf().get_val<uint64_t>("mgr_ttl_cache_expire_seconds");
+
+  PyFormatter pf;
+  PyJSONFormatter jf;
+  Formatter &f = ttl_seconds ? (Formatter&)jf : (Formatter&)pf;
```

Comment:
I had to do this hacky thing because only pyformatter and pyjsonformatter have a `PyObject* get` method. Formatter does not.
If you take a look at the end I have to call jf.get or pf.get. Also, it's not possible declaring these two formatters inside an if scope and then using it outside. 
Probably there is a better solution, and one of them would be refactoring pyformatter.
I'll add the comment.

Reply:
Upcast + later downcast with dynamic_cast?

Reply:
I tried downcasting but It won't work from Formatter to `PyFormatter` or `PyJSONFormatter`. Without having a refactoring I think this is the quickest way to do it.

Reply:
I tried with sample code and worked, but never mind, let's go with it!